### ISSUE TITLE
Logging in middleware cannot report client ID / expiry

### ISSUE THREAD

**Issue by petemoore** [taskcluster/taskcluster#3666]
**Describe the bug**
I _think_ the logging middleware will never report the clientId and client expiry of a request, since the `logRequest({ builder: this.builder, entry })` middleware entry is before the `remoteAuthentication({ signatureValidator, entry })` entry in `libraries/api/src/api.js`.

The logging middleware logs the clientId and client expiry if it is available, but I think it can never be, since at the point that code runs, the signature validator has not yet run, which is what determines which clientId made the request (if any) and what expiry it has (if any).
 
I haven't performed a thorough test of this hypothesis, as I'm heading into PTO, but wanted to log this thought before I go. I noticed that several of the tests in `libraries/api/test/logging_test.js` make a request with a client with valid credentials, but even those tests expect that no clientId is reported in the log. The fix may be as simple as switching the order of the middleware entries, as I think the middleware entries all call `next(err)` on error, rather than aborting, so I believe the full middleware stack is still executed even when an error occurs. But I haven't checked all the middleware entries, so this might not be universally true. Another problem might be if a later middleware entry relies on setup from the logging middleware entry - I haven't checked this either.

**Comment by djmitche** [comment]
This [does actually work](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22moz-fx-taskcluster-prod-4b87%22%0Aresource.labels.location%3D%22us-west1%22%0Aresource.labels.cluster_name%3D%22taskcluster-communitytc-v1%22%0Aresource.labels.namespace_name%3D%22communitytc-taskcluster%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22communitytc-taskcluster-taskcluster-1%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fpart-of%3D%22taskcluster%22%0AjsonPayload.Type%3D%22monitor.apiMethod%22%0AjsonPayload.Fields.clientId!%3D%22%22;timeRange=2020-10-19T12:16:40.407Z%2F2020-10-19T14:16:40.407Z?pageState=(%22savedViews%22:(%22i%22:%22d4b5b4fa2b26410d84b4fde2b70855b5%22,%22c%22:%5B%5D,%22n%22:%5B%5D))&authuser=0&project=moz-fx-taskcluster-prod-4b87).

![image](https://user-images.githubusercontent.com/28673/96463422-b4539100-11f4-11eb-9888-0dffa1b9821d.png)

Express middleware in general is a maze of nested functions.  In the case of the logging middleware:

https://github.com/taskcluster/taskcluster/blob/bfa495e06ce3f8cca315a7d8cfe2c5e4fc365287/libraries/api/src/middleware/logging.js#L48-L95

```javascript
const logRequest = ({ builder, entry }) => {
  return (req, res, next) => {
    const send = {};
    res.once('finish', send);
    res.once('close', send);
    next();
  };
};
```

so basically the handler just sets up some event listeners and calls through to the next middleware.  When those events fire (which should be after any auth is checked, or on an error), the logging function runs.
